container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go

```diff
@@ -1926,6 +2394,77 @@ func (manager *KubernetesManager) HasComputeNodes(ctx context.Context) (bool, er
 	return len(nodes.Items) != 0, nil
 }
 
+func (manager *KubernetesManager) AddLabelsToNode(ctx context.Context, nodeName string, labels map[string]string) error {
+	nodeClient := manager.kubernetesClientSet.CoreV1().Nodes()
+
+	node, err := nodeClient.Get(ctx, nodeName, globalGetOptions)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while trying to get node '%v'. Ensure node with name '%v' exists in cluster.", nodeName, nodeName)
+	}
+
+	// add node selectors to existing labels
+	for k, v := range labels {
+		node.Labels[k] = v
+	}
+
+	_, err = nodeClient.Update(ctx, node, metav1.UpdateOptions{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "",
+			APIVersion: "",
+		},
+		DryRun:          nil,
+		FieldManager:    "",
+		FieldValidation: "",
+	})
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while trying to update node '%v' with labels '%v'", nodeName, labels)
+	}
+
+	return nil
+}
+
+func (manager *KubernetesManager) RemoveLabelsFromNode(ctx context.Context, nodeName string, labels map[string]string) error {
```

Comment:
Same question as above about KT modifying node labels. E.g. if there's a bug in KT code, does that mean we blow away the labels for a node?

Reply:
yes, the `labels` provided would be removed. kurtosis needs to enforce only kurtosis labels are removed. as a protection against non kurtosis related labels being removed, i'll add a check to ensure all labels are prefixed with the kurtosis app id.